cvat-data/src/js/cvat-data.js

```diff
@@ -116,55 +119,95 @@ class FrameProvider {
         if (frameNumber in this._frames) {
             return this._frames[frameNumber];
         }
-
         return null;
     }
 
+
     /*
-        Method decodes a block of data
+        Method start asynchronic decode a block of data
 
         @param block - is a data from a server as is (ts file or archive)
         @param start {number} - is the first frame of a block
         @param end {number} - is the last frame of a block + 1
+        @param callback - callback)
+
     */
-    decode(block, start, end) {
-        async function decode(resolve, reject) {
-            // no more requests possible during one is running
+
+    StartDecode(block, start, end, needed_frame, callback)
```

Comment:
The first letter should be lowercase :) 
Classes are named with uppercase first letter in JS

Reply:
You don't need the argument "needed_frame" here now